### PR TITLE
Add linked_clone config option

### PR DESCRIPTION
**Pull request by bborysenko**
Allows to use linked clones with Vagrant 1.8

### CODE REVIEW

File: templates/Vagrantfile.erb [hunk 1]
```diff
@@ -73,6 +73,9 @@ Vagrant.configure("2") do |c|
      if config[:gui] == true || config[:gui] == false %>
     p.gui = <%= config[:gui] %>
 <%   end
+     if config[:linked_clone] == true || config[:linked_clone] == false %>
+    p.gui = <%= config[:linked_clone] %>
```
Comment (legal90) [inline templates/Vagrantfile.erb]: I guess, `p.gui` should be replaced with `p.linked_clone`:
```ruby
p.linked_clone = <%= config[:linked_clone] %>
```

Comment (bborysenko) [inline templates/Vagrantfile.erb]: fixed, @legal90 thanks

**Comment by lgeinert** [comment]
What is the process to cleanup the master after you are done using the linked clones?  kitchen-destroy only removes the dependent cloned kitchen disks.

**Comment by bborysenko** [comment]
Yes the master VM is not removed automatically by Vagrant and this should be done manually. But I guess kitchen-destroy has not to take care about that, since the final goal is just speed up each next test if run them in the same environment. For example when testing something locally or on the CI server.

**Comment by cheeseplus** [comment]
If some tests could be added to check the new options I'd be happy to look at merging.

**Comment by bborysenko** [comment]
@cheeseplus done

**Comment by bborysenko** [comment]
Fixed typo and rebased

**Comment by jeremiahsnapp** [comment]
@cheeseplus Would you mind looking at merging this again. I'd like to see this feature get added if possible. Thanks!

**Comment by bborysenko** [comment]
Thanks to @legal90 now with `parallels` provider support.

**Comment by cheeseplus** [comment]
:+1:

**Comment by smurawski** [comment]
:+1:

**Pull request merged**